src/waitForElements.js

```diff
@@ -0,0 +1,43 @@
+function waitForElements(
+  callback = () => {},
+  {
+    container = document,
+    timeout = 4500,
+    mutationObserverOptions = {subtree: true, childList: true},
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    // Disabling eslint prefer-const below: either prefer-const or no-use-before-define triggers.
+    let lastError, observer, timer // eslint-disable-line prefer-const
```

Comment:
Really odd that both of these rules trigger 🤔 I don't see how this code is in violation. An ESLint bug?

Reply:
If I have it as `const`, I have to initialize it at definition, i.e. at this line:
```
timer = setTimeout(onTimeout, timeout)
```
but then:
```
    function onDone(error, result) {
      clearTimeout(timer)
```
is a usage before definition of `timer`